changes/2020-07-13_detect-base64-encoded-messages/change.md

```diff
@@ -0,0 +1,88 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Detect Base64-encoded Messages
+
+## Affected Features
+
+This serves as a reference of all features that this change affects.
+
+| Feature                                                                                                                                     |
+| ------------------------------------------------------------------------------------------------------------------------------------------- |
+| [Decrypt](https://github.com/awslabs/aws-encryption-sdk-specification/blob/623992d999db0b309d8a8adbd664f0d72feee813/client-apis/decrypt.md) |
+|                                                                                                                                             |
+
+## Affected Specifications
+
+This serves as a reference of all specification documents that this change affects.
+
+| Specification                           |
+| --------------------------------------- |
+| [Decrypt](../../client-apis/decrypt.md) |
+
+## Affected Implementations
+
+This serves as a reference for all implementations that this change affects.
+
+| Language   | Repository                                                                            |
+| ---------- | ------------------------------------------------------------------------------------- |
+| C          | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Java       | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-java)                |
+| Javascript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+| Python     | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-python)     |
+
+## Definitions
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+When a decryption attempt is made on the Base64-encoding of a valid message,
```

Comment:
> a decryption attempt is made on the Base64-encoding of a valid message

Does this mean a user is trying to decrypt a Base64-string, the utf-8 encoding of a Base64-string, or both/either?

Reply:
The UTF-8 encoding of the string - decrypt accepts a message as a byte sequence after all.

FWIW, a Base64 string only uses ASCII characters so nearly all encodings will use the same byte values.

Reply:
> decrypt accepts a message as a byte sequence after all

The spec must acknowledge dynamically typed languages though, no?

Reply:
Absolutely! But it MUST interpret whatever input it is given as a sequence of bytes somehow. If it's idiomatic to accept a string because it quacks like a byte sequence, the operation has to be clear about how it will interpret it as bytes: whether it will use a particular hard-coded encoding like UTF-8, or rely on an encoding attached to the string value, etc. 

In a particular language implementation, I would expect the documentation to cover this if it's not obvious by context.